src/qml/qgismobileapp.qml

```diff
@@ -2379,6 +2388,94 @@ ApplicationWindow {
         screenLocker.enabled = true
       }
     }
+
+    MenuSeparator {
+      width: parent.width
+      visible: canvasMenuFeatureListInstantiator.count > 0
+    }
+
+    Instantiator {
+      id: canvasMenuFeatureListInstantiator
+
+      model: MultiFeatureListModel {
+        id: canvasMenuFeatureListModel
+      }
+
+      Menu {
+        id: featureMenu
+
+        property int fid: featureId
+        property var featureLayer: currentLayer
+
+        topMargin: sceneTopMargin
+        bottomMargin: sceneBottomMargin
+
+        title: layerName + ': ' + featureName
+        font: Theme.defaultFont
+        leftPadding: 10
+
+        width: {
+            var result = 0;
+            var padding = 0;
+            for (var i = 0; i < count; ++i) {
+                var item = itemAt(i);
+                result = Math.max(item.contentItem.implicitWidth, result);
+                padding = Math.max(item.padding, padding);
+            }
+            return Math.min(result + padding * 2,mainWindow.width - 20);
+        }
+
+        Component.onCompleted: {
+          if (featureMenu.icon !== undefined) {
+            featureMenu.icon.source = Theme.getThemeVectorIcon('ic_info_white_24dp')
+          }
+        }
+
+        MenuItem {
+          text: qsTr('Open Feature Form')
+          font: Theme.defaultFont
+          icon.source: Theme.getThemeIcon( "ic_baseline-list_alt-24px" )
+          leftPadding: 10
+
+          onTriggered: {
+            featureForm.model.setFeatures(menu.featureLayer, '$id = ' + menu.fid)
+            featureForm.selection.focusedItem = 0
+            featureForm.state = "FeatureForm"
+          }
+        }
+
+        MenuSeparator { width: parent.width }
+
+        MenuItem {
+          text: qsTr('Duplicate Feature')
+          font: Theme.defaultFont
+          enabled: projectInfo.insertRights
+          icon.source: Theme.getThemeVectorIcon( "ic_duplicate_black_24dp" )
+          leftPadding: 10
+
+          onTriggered: {
+            featureForm.model.setFeatures(menu.featureLayer, '$id = ' + menu.fid)
+            featureForm.selection.focusedItem = 0
+            featureForm.multiSelection = true
+            featureForm.selection.toggleSelectedItem(0)
+            featureForm.state = "FeatureList"
+            if (featureForm.model.canDuplicateSelection) {
+              if (featureForm.selection.model.duplicateFeature(featureForm.selection.focusedLayer,featureForm.selection.focusedFeature)) {
+                displayToast(qsTr("Successfully duplicated feature"))
+
+                featureForm.selection.focusedItem = -1
+                moveFeaturesToolbar.initializeMoveFeatures()
+                return;
+              }
+            }
+            displayToast(qsTr( "Feature duplication not available" ))
```

Comment:
Small style comment, it would be great if at least each file and PR is sticking to one quote and spacing convention.

E.g. this line is 
```
displayToast(qsTr( "Feature duplication not available" ))
```
but few lines above we have:
```
text: qsTr('Duplicate Feature')
```

Reply:
@suricactus , corrected.